src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/AssigningSymbolAndItsMemberInSameStatement.Fixer.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Text;
+
+#nullable enable
+
+namespace Microsoft.CodeQuality.Analyzers.QualityGuidelines
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic, Name = AssigningSymbolAndItsMemberInSameStatement.RuleId), Shared]
+    public abstract class AssigningSymbolAndItsMemberInSameStatementFixer : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(AssigningSymbolAndItsMemberInSameStatement.RuleId);
+
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            // For something like `a.x = a = b;`, we offer 3 code fixes:
+            // First:
+            //     a = b;
+            //     a.x = b;
+            // Second:
+            //     a = b;
+            //     a.x = a;
+            // Third:
+            //     var temp = a;
+            //     a = b;
+            //     temp.x = b;
+
+            var title = MicrosoftCodeQualityAnalyzersResources.AssigningSymbolAndItsMemberInSameStatementTitle;
+            context.RegisterCodeFix(new MyCodeAction(title,
+                 async ct => await SplitAssignment(context.Document, context.Span, ct).ConfigureAwait(false),
+                 equivalenceKey: title),
+            context.Diagnostics);
+            return Task.CompletedTask;
+        }
+
+        private async Task<Document> SplitAssignment(Document document, TextSpan span, CancellationToken cancellationToken)
+        {
+            // This method splits `a.x = a = b` to:
+            // a = b;
+            // a.x = b;
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+            // a.x = a = b
+            var parentAssignment = root.FindNode(span).Parent;
+
+            if (!TryGetAssignmentExpressionParts(parentAssignment, out _, out var right))
+            {
+                return document;
+            }
+
+            if (!TryGetAssignmentExpressionParts(right, out _, out var rightOfRight))
+            {
+                return document;
+            }
+
+            // a = b;
+            right = right.WithTriviaFrom(parentAssignment);
+
+            // a.x = b;
+            var firstEqualsLastAssignment = GetAssignmentWithRight(parentAssignment, rightOfRight);
+
+            root = root.ReplaceNode(parentAssignment, new[] { right, firstEqualsLastAssignment });
```

Comment:
@Evangelink I re-wrote this without using the `DocumentEditor`. But it's still not working:

This line is failing with:

> System.InvalidOperationException : The item specified is not the element of a list.

1. The exception is unclear from Roslyn's side.
2. Assuming that this exception means that the node isn't part of the tree, I can't see any reason for that.

Any idea?

Pinging @sharwell and @mavasani as well.